crates/compiler/test_gen/src/helpers/llvm.rs

```diff
@@ -286,13 +284,27 @@ fn create_llvm_module<'a>(
         );
     };
 
-    if main_fn.verify(true) {
-        function_pass.run_on(&main_fn);
-    } else {
```

Comment:
I don't fully understand this change. I believe LLVM optimisation no longer occurs at a function level and so the function_pass manager is no longer required. Here I have removed this, but we still keep the verification check. 

Request this be looked at this more closely.